packages/client/src/internals/dataLoader.ts

```diff
@@ -26,100 +37,125 @@ type BatchLoader<TKey, TValue> = {
 export function dataLoader<TKey, TValue>(
   batchLoader: BatchLoader<TKey, TValue>,
 ) {
-  let batch: Batch<TKey, TValue> | null = null;
-  let dispatchTimer: NodeJS.Timer | number | null = null;
+  let pendingItems: BatchItem<TKey, TValue>[] | null = null;
```

Comment:
Switched this from being a list of items rather than a batch. The batch[es] is/are created in `dispatch`